cis194/week4/tien/homework4/src/Homework4.hs

```diff
@@ -0,0 +1,30 @@
+module Homework4 where
+
+-- Exercise 1 --
+-- 1.1 --
+fun1 :: [Integer] -> Integer
+fun1 [] = 1
+fun1 (x:xs)
+  | even x = (x - 2) * fun1 xs
+  | otherwise = fun1 xs
+
+fun1' :: [Integer] -> Integer
+fun1' = product . map (subtract 2) . filter (even)
+
+-- 1.2 --
+fun2 :: Integer -> Integer
+fun2 1 = 0
+fun2 n
+  | even n = n + fun2 (n `div` 2)
+  | otherwise = fun2 (3 * n + 1)
+
+fun2' :: Integer -> Integer
+fun2' =
+  sum .
+  filter even .
+  takeWhile (> 1) .
+  iterate
+    (\x ->
+       if even x
+         then x `div` 2
+         else 3 * x + 1)
```

Comment:
In Haskell, we usually put symbols at the beginning of a line so a more conventional style is.
```Haskell
fun2' 
    = sum 
    . filter even 
    . takeWhile (> 1)
    . iterate (\x ->
        if even x
        then x `div` 2 -- also align then & else with if
        else 3 * x + 1)
```

Reply:
Ha this is auto-formatted by the plugin. Maybe i will try if I can config the alignment. Thanks.